File: app/src/main/java/life/plank/juna/zone/data/network/model/FeedItem.java
package life.plank.juna.zone.data.network.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

import lombok.Data;

/**
 * Created by plank-arfaa on 06/02/18.
 */

@Data
public class FeedItem {
    private String id;
    private String title;
    private String url;
    private String source;
    private String datePublished;
    private String summary;
    private Thumbnail thumbnail;
    private String dateCreated;
    private String contentType;
    private Interaction interactions;
    private String description;
    @SerializedName("actor")
    @Expose
    private User user;
    List<FeedItemComment> comments;
    /** Field to store the tile width of the feed item */
    private int tileWidth;
    /**
     * Field to check whether the feed item is pinned
     * TODO: Remove once isPinned is provided in Interactions
     */
    private boolean isPinned;
    private String pinId;
    private int previousPosition;
}

File: app/src/main/java/life/plank/juna/zone/data/model/feed/FeedItem.kt
package life.plank.juna.zone.data.model.feed

import android.os.Parcelable
import com.google.gson.annotations.*
import kotlinx.android.parcel.*
import life.plank.juna.zone.data.model.board.Interaction
import life.plank.juna.zone.data.model.user.User
import life.plank.juna.zone.util.common.AppConstants.*
import java.util.*
import kotlin.collections.ArrayList

@Parcelize
data class FeedItem(
        var id: String? = "",
        var title: String,
        var url: String? = null,
        var source: String? = null,
        var summary: String? = null,
        var thumbnail: Thumbnail? = Thumbnail(),
        var dateCreated: Date? = null,
        var contentType: String,
        var backgroundColor: MutableList<String>? = ArrayList(),
        var interactions: Interaction = Interaction(),
        var description: String? = null,
        var comments: @RawValue List<FeedItemComment>? = null,
        @SerializedName("actor") @Expose var user: User? = null
) : Parcelable {

    constructor() : this("", "", "", "", "", Thumbnail(), Date(), "", null, Interaction())

    /**
     * Constructor for WebLink Posts
     */
    constructor(title: String, url: String?, source: String?, summary: String?, description: String?, imageUrl: String) : this(
            title = title,
            url = url,
            source = source,
            summary = summary,
            description = description,
            thumbnail = Thumbnail(imageUrl),
            contentType = NEWS
    )

    /**
     * Constructor for Text Posts
     */
    constructor(title: String, backgroundColor: MutableList<String>?) : this(
            title = title,
            backgroundColor = backgroundColor,
            contentType = ROOT_COMMENT
    )
}